### PR TITLE
UI: Updates search to search resources by tags

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes
- If user types directly in the search box,then resources
  search by it's name, displayName and tags, if keyword like
  `tags:<tagName>,<tagName>` is typed to search 
  then it search by resources's tags only
    
- Adds searched tag to url query params as like `tag=tag1&tag=tag2`
    
Signed-off-by: Shiv Verma <shverma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by PuneetPunamiya** [comment]
@pratap0007  could you please rebase this pr ?

**Review by PuneetPunamiya — COMMENTED** [review]
Thanks 🤙🏻 
/approve
/meow
/woof

File: ui/src/store/resource.ts [hunk 1]
```diff
@@ -150,6 +155,9 @@ export const ResourceStore = types
     },
     setURLParams(url: string) {
       self.urlParams = url;
+    },
+    setSearchedTags(tags: Array<string>) {
```
[the remark this answers is not in the document]
Comment (pratap0007) [reply ui/src/store/resource.ts:159]: both data type declaration are same but `Array<string>` is a more generic type

**Pull request merged**